packages/protocol-http/src/Field.ts

```diff
@@ -0,0 +1,59 @@
+import { FieldPosition } from "./FieldPosition";
+
+export type FieldOptions = { 
+  name: string;
+  values?: string[];
+  kind?: FieldPosition
+};
+
+/**
+ * A name-value pair representing a single field
+ * transmitted in an HTTP Request or Response.
+ * 
+ * The kind will dictate metadata placement within
+ * an HTTP message.
+ * 
+ * All field names are case insensitive and
+ * case-variance must be treated as equivalent.
+ * Names MAY be normalized but SHOULD be preserved
+ * for accuracy during transmission.
+ */
+export class Field {
+  public name: string;
+  public values: string[];
+  public kind: FieldPosition;
+
+  constructor(options: FieldOptions) {
+    this.name = options.name;
+    this.values = options.values ?? [];
+    this.kind = options.kind ?? FieldPosition.HEADER;
+  }
+
+  // Appends a value to the field.
+  public add(value: string): void {
+    this.values.push(value);
+  }
+
+  // Overwrite existing field values.
+  public set(values: string[]): void {
+    this.values = values;
+  }
+
+  // Remove all matching entries from list.
+  public remove(value: string): void {
+    this.values = this.values.filter((v) => v !== value);
+  }
+
+  // Get comma-delimited string
+  // Values with spaces or commas MUST be double-quoted
+  public toString(): string {
+    return this.values
+      .map((v) => (v.includes(",") || v.includes(" ") ? `"${v}"` : v))
+      .join(", ");
```

Comment:
is the space after comma necessary?

Reply:
Leaving the spaces, as based on discussion offline, multi-value headers can have spaces